api/client/proto/types.go

```diff
@@ -0,0 +1,14 @@
+package proto
+
+import "github.com/gravitational/trace"
+
+// CheckAndSetDefaults checks and sets default values
+func (req *GenerateServerKeysRequest) CheckAndSetDefaults() error {
+	if req.HostID == "" {
+		return trace.BadParameter("missing parameter HostID")
+	}
+	if len(req.Roles) != 1 {
```

Comment:
Should this be `<1`?  If not then why do we take a list?

Reply:
For host certs we always have a single role (node, auth, proxy, etc.)

I'm not sure why the request takes a slice, it seems to have always been that way. This change is not new code, it's just moving `CheckAndSetDefaults` from the old `auth.GenerateServerKeysRequest` to the new `protocol.GenerateServerKeysRequest`.

Reply:
Right. Maybe leave a TODO to clean this later? Seems strange.

Reply:
I will just change this to a single role and not a list. Easier to do now than later.